src/methods/function.js

```diff
@@ -1151,23 +1142,26 @@ export function incorporateSavedCompletion(realm: Realm, c: void | Completion |
       realm.applyEffects(joined_effects);
       let jc = joined_effects[0];
       invariant(jc instanceof AbruptCompletion);
-      throw jc;
+      return jc;
     }
   }
   return c;
 }
 
 export function EvaluateStatements(
   body: Array<BabelNodeStatement>,
-  blockValue: void | NormalCompletion | Value,
+  initialBlockValue: void | Value,
   strictCode: boolean,
   blockEnv: LexicalEnvironment,
   realm: Realm
-): NormalCompletion | Value {
+): Value {
+  let context = realm.getRunningContext();
+  let blockValue = initialBlockValue;
   for (let node of body) {
     if (node.type !== "FunctionDeclaration") {
-      let res = blockEnv.evaluateAbstractCompletion(node, strictCode);
-      invariant(!(res instanceof Reference));
+      let res = blockEnv.evaluateAbstractCompletion(node, strictCode); //todo: call evaluateCompletion
```

Comment:
This is needed to avoid early termination in the case of node evaluating to joined abrupt completion.